index.js

```diff
@@ -319,53 +341,18 @@ class Grc extends Base {
       timeout: 120000
     }, _opts)
 
-    if (_.isFunction(_cb)) {
-      isPromise = false
-      peer.request(service, {
-        action,
-        args
-      }, opts, cb)
-
-      return
-    }
-
-    isPromise = true
-    return new Promise((resolve, reject) => {
-      _resolve = resolve
-      _reject = reject
-      peer.request(service, {
-        action,
-        args
-      }, opts, cb)
-    })
-  }
-
-  map (service, action, args, _opts = {}, _cb) {
-    if (!_.isString(action)) return _cb(new Error('ERR_GRC_REQ_ACTION_INVALID'))
-    if (!_.isArray(args)) return _cb(new Error('ERR_GRC_REQ_ARGS_INVALID'))
-    if (!_.isFunction(_cb)) return _cb(new Error('ERR_GRC_REQ_CB_INVALID'))
-
-    let isExecuted = false
-
-    const cb = (err, res) => {
-      if (isExecuted) {
-        console.error('ERR_DOUBLE_CB', service, action, JSON.stringify(args))
-        return
-      }
-      isExecuted = true
-      if (err === 'ERR_TIMEOUT') {
-        console.error('ERR_TIMEOUT received', service, action)
-      }
-      _cb(err ? new Error(err) : null, res)
+    if (isPromise) {
+      return new Promise((resolve, reject) => {
+        _resolve = resolve
+        _reject = reject
+        peer[reqType](service, {
+          action,
+          args
+        }, opts, cb)
+      })
     }
 
-    const peer = service.indexOf('sec:') === 0 ? this.peerSec : this.peer
```

Comment:
why this was removed?

Reply:
it's done [in _doReq here](https://github.com/bitfinexcom/bfx-facs-grc/pull/15/files/09ad7687089afdeb12e37411b9f4547aef4648db#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R338)